### PR TITLE
ci: run interop tests on bigger hardware

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

With the move to running our CI on self-hosted runners, this job tends to always be the slowest one. We move it to a bigger runner such that it finishes closer to all the other jobs.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: .github/workflows/interop-test.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ concurrency:
 jobs:
   run-multidim-interop:
     name: Run multidimensional interoperability tests
-    runs-on: ${{ fromJSON(github.repository == 'libp2p/rust-libp2p' && '["self-hosted", "linux", "x64", "xlarge"]' || '"ubuntu-latest"') }}
+    runs-on: ${{ fromJSON(github.repository == 'libp2p/rust-libp2p' && '["self-hosted", "linux", "x64", "2xlarge"]' || '"ubuntu-latest"') }}
```
Comment (thomaseizinger) [inline .github/workflows/interop-test.yml]:
```suggestion
    runs-on: ${{ fromJSON(github.repository == 'libp2p/rust-libp2p' && '["self-hosted", "linux", "x64", "4xlarge"]' || '"ubuntu-latest"') }}
```

**Comment by mxinden** [comment]
Fine by me, though not familiar with the setup. Will defer to @galargh. Thanks for making our lives easier @thomaseizinger and @galargh!

**Comment by galargh** [comment]
It'd be great if we could merge https://github.com/libp2p/test-plans/pull/154 first. With that in place we'd be able to actually make use of the resources that we get on a bigger machine.

**Comment by thomaseizinger** [comment]
> It'd be great if we could merge https://github.com/libp2p/test-plans/pull/154 first. With that in place we'd be able to actually make use of the resources that we get on a bigger machine. 

Sounds good to me! From my end that looks good to go in. Perhaps nudge Marco again to see what needs to be done to land it.

**Review by galargh — APPROVED** [review]

**Pull request merged**